pyClient/test/test_contract_base_mixer.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2015-2019 Clearmatics Technologies Ltd
+#
+# SPDX-License-Identifier: LGPL-3.0+
+
+import test_commands.mock as mock
+import zeth.constants as constants
+import zeth.joinsplit
+
+from typing import Any
+
+# Mixer variables
+size_value = 64  # pylint: disable=no-member,invalid-name
+digest_length = 256  # pylint: disable=no-member,invalid-name
+field_capacity = 253  # pylint: disable=no-member,invalid-name
+js_in = 2  # pylint: disable=no-member,invalid-name
+js_out = 2  # pylint: disable=no-member,invalid-name
+packing_residue_length = digest_length % field_capacity \
+    # pylint: disable=no-member,invalid-name
+if digest_length <= field_capacity:
+    packing_residue_length = 0
+# offset = 2 * size_value + packing_residue_length
+length_bit_residual = 2 * size_value + packing_residue_length * \
+    (1 + 2 * js_in + js_out)  # pylint: disable=no-member,invalid-name
+# nb_field_residual = math.ceil(length_bit_residual / field_capacity)
+
+# The variable inputs represents a dummy primary input array,
+# it is structured as follows,
+# inputs =
+#   rt || {sn}_1,2 || {cm}_1,2 || h_sig || {h}_1,2 || residual_bits
+# residual_bits =
+#   v_in || v_out || h_sig || {sn}_1,2  || {cm}_1,2  || {h}_1,2
+# We set dummy values for all variables but residual_bits as follows:
+# residual_bits = 713623846352979940490457358497079434602616037, or in bits
+# 1-4:   00000000 00000000 00000000 00000000
+# 5-8:   00000000 00000000 00000000 00000000
+# 9-12:  00000000 00000000 00000000 00000000
+# 13-16: 00000000 00011111 11111111 11111111
+# 17-20: 11111111 11111111 11111111 11111111
+# 21-24: 11111111 11100000 00000000 00000000
+# 25-28: 00000000 00000000 00000000 00000000
+# 29-32: 00000000 00011100 00010100 11100101
+# This corresponds to
+# v_in  = 0xFFFFFFFFFFFFFFFF
+# v_out = 0x0000000000000000
+# h_sig = 111
+# sn_0  = 000
+# sn_1  = 001
+# cm_0  = 010
+# cm_1  = 011
+# h_0   = 100
+# h_1   = 101
+# The values were set to be easily distinguishable.
+inputs = [
+    0,  # root
+    1,  # sn_0
+    1,  # sn_1
+    2,  # cm_0
+    2,  # cm_1
+    3,  # h_sig
+    4,  # h_0
+    4,  # h_1
+    713623846352979940490457358497079434602616037] \
+        # residual bits ; pylint: disable=no-member,invalid-name
+
+
+def test_extract(mixer_instance: Any) -> int:
+    # This test checks that we extract the right values
+    # for {sn}, {cm} and {h} regardless of whether
+    # the bits are located in the same byte
+    # or two different ones (c.f. residual_bits above).
+    print("--- testing ", "test_extract")
+
+    start = 0
+    for i in range(2*js_in+js_out):
+        res = mixer_instance.functions.extract_extra_bits(
+            start,
+            packing_residue_length,
+            inputs
+            ).call()
+        if int.from_bytes(res, byteorder="big") != i:
+            print("ERROR: extracted wrong value")
+            return 0
```

Comment:
Why return `0`? 0 indicates a success, but you have an error message here. It should be =/= 0

Reply:
I am used to returning 1 if the test passes, 0 otherwise.
Will update to "0=pass" convention.

Reply:
Fixed